### PR TITLE
library: Adds ColorPicker Demo

### LINKED ISSUE #334: Add ColorPicker Demo
https://libportal.org/method.Portal.pick_color.html

### PR DESCRIPTION
**Pull request by SoNiC-HeRE**
[Screencast from 2023-06-14 11-47-15.webm](https://github.com/sonnyp/Workbench/assets/96797205/293c46d0-fead-4724-aa38-02e5be64514c)

Closes #334

### CODE REVIEW

**Review by sonnyp — CHANGES_REQUESTED** [review]
You're not using the promisified version using `await`

Please refer to the `Screenshot` library entry.

File: src/Library/demos/Color Picker/main.blp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.StatusPage {
+  title: "Color Picker";
```
Comment (sonnyp) [inline src/Library/demos/Color Picker/main.blp:5]:
```suggestion
  title: _("Color Picker");
```

File: src/Library/demos/Color Picker/main.blp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  Box {
+    orientation: vertical;
+    halign: center;
+
+    Button button {
+      label: _("Select Color");
+      margin-bottom: 42;
+      halign: center;
+      styles ["suggested-action"]
```
Comment (sonnyp) [inline src/Library/demos/Color Picker/main.blp:17]:
```suggestion
      styles ["suggested-action", "pill"]
```

File: src/Library/demos/Color Picker/main.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    const colorInfo = color.print(false);
+
+    // The picked color is a GVariant of the form (ddd), containing red, green and blue components in the range [0,1].
+    console.log(`Selected Color is: ${colorInfo}`);
+  } catch (err) {
+    logError(err);
+  }
+}
+
+button.connect("clicked", pickColor);
```
Comment (sonnyp) [inline src/Library/demos/Color Picker/main.js:26]: Similar to https://github.com/sonnyp/Workbench/pull/330#discussion_r1231484336

Remove the try/catch, rename and do

```suggestion
button.connect("clicked", () => {
  onClicked().catch(logError);
});
```

File: src/Library/demos/Color Picker/main.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+const portal = new Xdp.Portal();
+const parent = XdpGtk.parent_new_gtk(workbench.window);
+const button = workbench.builder.get_object("button");
+
+Gio._promisify(Xdp.Portal.prototype, "pick_color", "pick_color_finish");
+
+async function pickColor() {
+  try {
+    const color = await portal.pick_color(parent, null);
```
Comment (sonnyp) [inline src/Library/demos/Color Picker/main.js:16]: the result of `pick_color` isn't really a color, I would naming the return value `result` instead and move the comment there

```suggestion
    // result is a GVariant of the form (ddd), containing red, green and blue components in the range [0,1]
    const result = await portal.pick_color(parent, null);
```

File: src/Library/demos/Color Picker/main.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    const colorInfo = color.print(false);
+
+    // The picked color is a GVariant of the form (ddd), containing red, green and blue components in the range [0,1].
+    console.log(`Selected Color is: ${colorInfo}`);
```
Comment (sonnyp) [inline src/Library/demos/Color Picker/main.js:20]: wdyt we create a rgba color with https://docs.gtk.org/gdk4/struct.RGBA.html and then use `color.to_string()` ?

similar to the Color Dialog entry
You can read rgb from `result` using `const [r,g,b] = result.deepUnpack()`

**Review by sonnyp — CHANGES_REQUESTED** [review]

File: src/Library/demos/Color Picker/main.js (hunk 5 above)
Comment (SoNiC-HeRE) [reply src/Library/demos/Color Picker/main.js:20]: Sure

**Comment by SoNiC-HeRE** [comment]
Made the following changes:
- Removed async handler
- Refined Code
- Used Gdk.RGBA for better output

**Comment by SoNiC-HeRE** [comment]
Made the following changes:
- Removed async handler
- Refined Code
- Used Gdk.RGBA for better output

**Review by sonnyp — APPROVED** [review]
Well done :sparkles:

**Pull request merged**